packages/mui-system/src/style.js

```diff
@@ -35,6 +35,14 @@ export function getStyleValue(themeMapping, transform, propValueFinal, userValue
     value = getPath(themeMapping, propValueFinal) || userValue;
   }
 
+  if (typeof value === 'object') {
+    if (process.env.NODE_ENV !== 'production') {
+      console.warn(
+        `MUI: The value found in theme for prop: "${propValueFinal}" is an [Object] instead of string or number. Check if you forgot to add the correct dotted notation, eg, "background.paper" instead of "background".`,
+      );
+    }
```

Comment:
Please also set the `userValue` to `value` here like you were doing last time. If a proper value is not found in theme, it should just use whatever user provided in prop.

Reply:
Actually, it was removed on purpose because there are cases when `value` could be an object such as Typography theme key body1 or h1 which is assigned as an object through handlebreakpoints. So, I think just warning the user for possible mistake is enough. If my point is unclear kindly refer to this [test case](https://github.com/mui/material-ui/blob/de193497e41ec34fae0512c9e8f4fe3298d8ce7e/packages/mui-system/src/styleFunctionSx/styleFunctionSx.test.js#L96)